src/Stratis.Features.FederatedPeg/TargetChain/SignedMultisigTransactionBroadcaster.cs

```diff
@@ -57,38 +64,36 @@ public class SignedMultisigTransactionBroadcaster : ISignedMultisigTransactionBr
 
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
         }
 
-        private async Task OnCrossChainTransactionFullySignedAsync(CrossChainTransferTransactionFullySigned @event)
+        /// <inheritdoc />
+        public async Task<SignedMultisigTransactionBroadcastResult> BroadcastFullySignedTransfersAsync()
         {
             if (this.ibdState.IsInitialBlockDownload() || !this.federationWalletManager.IsFederationWalletActive())
             {
-                this.logger.LogInformation("Federation wallet isn't active or in IBD. Not attempting to broadcast signed transactions.");
-                return;
+                this.logger.LogInformation("Federation wallet isn't active or the node is in IBD.");
+                return new SignedMultisigTransactionBroadcastResult() { Message = "The federation wallet isn't active or the node is in IBD." };
             }
 
-            TxMempoolInfo txInfo = await this.mempoolManager.InfoAsync(@event.Transfer.PartialTransaction.GetHash()).ConfigureAwait(false);
-            if (txInfo != null)
+            ICrossChainTransfer[] fullySignedTransfers = this.crossChainTransferStore.GetTransfersByStatus(new[] { CrossChainTransferStatus.FullySigned });
+
+            if (fullySignedTransfers.Length == 0)
             {
-                this.logger.LogDebug("Deposit ID '{0}' already in the mempool.", @event.Transfer.DepositTransactionId);
-                return;
+                this.logger.LogInformation("There are no fully signed transactions to broadcast.");
+                return new SignedMultisigTransactionBroadcastResult() { Message = "There are no fully signed transactions to broadcast." };
             }
 
-            this.logger.LogDebug("Broadcasting deposit-id={0} a signed multisig transaction {1} to the network.", @event.Transfer.DepositTransactionId, @event.Transfer.PartialTransaction.GetHash());
+            var result = new SignedMultisigTransactionBroadcastResult();
 
-            await this.broadcasterManager.BroadcastTransactionAsync(@event.Transfer.PartialTransaction).ConfigureAwait(false);
-
-            // Check if transaction was actually added to a mempool.
-            TransactionBroadcastEntry transactionBroadCastEntry = this.broadcasterManager.GetTransaction(@event.Transfer.PartialTransaction.GetHash());
-
-            if (transactionBroadCastEntry?.TransactionBroadcastState == TransactionBroadcastState.CantBroadcast && !CrossChainTransferStore.IsMempoolErrorRecoverable(transactionBroadCastEntry.MempoolError))
+            foreach (ICrossChainTransfer fullySignedTransfer in fullySignedTransfers)
             {
-                this.logger.LogWarning("Deposit ID '{0}' rejected due to '{1}'.", @event.Transfer.DepositTransactionId, transactionBroadCastEntry.ErrorMessage);
-                this.store.RejectTransfer(@event.Transfer);
+                result.Items.Add(await BroadcastFullySignedTransfersAsync(fullySignedTransfer));
```

Comment:
This could be changed to conditionally broadcast depending on whether the transaction is already in mempool.